src/Database/Connection.php

```diff
@@ -237,13 +297,24 @@ public function getDisconnectRetry(): CommandRetry
     }
 
     /**
-     * Gets the driver instance.
+     * Returns the primary connection role driver.
      *
      * @return \Cake\Database\DriverInterface
      */
     public function getDriver(): DriverInterface
```

Comment:
I would just add the `$role` argument this method and by default return the primary. A separate `getRoleDriver()` seems unnecessary.

Reply:
This is the part where 4.next breaks down.

Reply:
What breaks down? We're allowed to add new parameters to methods if they have default values. The `getDriver()` method isn't in any interfaces.

Reply:
I thought we couldn't change the signature of public functions at all.

Reply:
Ok. I had not seen that list before. I thought public functions were immutable.